morseMessenger/flashHook.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useCallback } from 'react';
+import { morseDictionary, DictOptions } from './morseDictionary';
+
+const TORCH_ON = 'torch';
+const TORCH_OFF = 'off';
+const PAUSE = 1000;
+const FLASH_PAUSE = 400;
+const FLASH_SHORT = 100;
+const FLASH_LONG = 700;
+
+export default function useFlash() {
+  const [flashMode, setFlashMode] = useState(TORCH_OFF);
+
+  function torchOffAfter(timeInMs: number) {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        setFlashMode(TORCH_OFF);
+        resolve();
+      }, timeInMs);
+    });
+  }
+  function torchOnAfter(timeInMs: number) {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        setFlashMode(TORCH_ON);
+        resolve();
+      }, timeInMs);
+    });
+  }
+
+  function short(): Promise<any> {
+    return new Promise((resolve, reject) => {
+      torchOnAfter(FLASH_PAUSE)
+        .then(() => torchOffAfter(FLASH_SHORT))
+        .then(resolve);
+    });
+  }
+  function long() {
+    return new Promise((resolve, reject) => {
+      torchOnAfter(FLASH_PAUSE)
+        .then(() => torchOffAfter(FLASH_LONG))
+        .then(resolve);
+    });
+  }
+  function pause() {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve();
+      }, PAUSE);
+    });
+  }
+
+  function translateToMorse(sentence: string) {
```

Comment:
again -> you should extract it back to a separate file.

Otherwise you are always creating it, so `useCallback` will always create a new function.